src/elastio-snap.c

```diff
@@ -548,17 +548,28 @@ static int elastio_snap_should_remove_suid(struct dentry *dentry)
 	#define vzalloc(size) __vmalloc(size, GFP_KERNEL | __GFP_HIGHMEM | __GFP_ZERO, PAGE_KERNEL)
 #endif
 
-#if !defined HAVE_MAKE_REQUEST_FN_IN_QUEUE && defined HAVE_BDOPS_SUBMIT_BIO
-	// Linux kernel version 5.9+
+#if !defined HAVE_MAKE_REQUEST_FN_IN_QUEUE && defined HAVE_BDOPS_SUBMIT_BIO_UINT
+	// Linux kernel version 5.9 - 5.15
 	// make_request_fn has been moved from the request queue structure to the
-	// block_device_operations as submit_bio function.
+	// block_device_operations as submit_bio function with UINT return type.
 	// See https://github.com/torvalds/linux/commit/c62b37d96b6eb3ec5ae4cbe00db107bf15aebc93
 	#define USE_BDOPS_SUBMIT_BIO
 
 	// Prototype bdev->fops->submit_bio but with the name already used in the code
 	typedef blk_qc_t (make_request_fn) (struct bio *bio);
 #endif
 
+#if !defined HAVE_MAKE_REQUEST_FN_IN_QUEUE && defined HAVE_BDOPS_SUBMIT_BIO
+	// Linux kernel version 5.16+
+	// make_request_fn has been moved from the request queue structure to the
+	// block_device_operations as submit_bio function with VOID return type.
+	// See https://github.com/torvalds/linux/commit/c62b37d96b6eb3ec5ae4cbe00db107bf15aebc93
```

Comment:
Let's reflect new change in the comment with the appropriate link to a new commit in the Linux kernel.
```suggestion
	// Linux kernel version 5.16+
	// submit_bio function in the block_device_operations structure has changed its return type to VOID.
	// See https://github.com/torvalds/linux/commit/3e08773c3841e9db7a520908cc2b136a77d275ff#diff-79b436371fdb3ddf0e7ad9bd4c9afe05160f7953438e650a77519b882904c56bR1181
```